nav2_collision_monitor/src/source_base.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/source_base.hpp"
+#include "nav2_collision_monitor/kinematics.hpp"
+
+#include <exception>
+
+#include "tf2/convert.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+
+#include "nav2_util/node_utils.hpp"
+
+namespace nav2_collision_monitor
+{
+
+SourceBase::SourceBase(
+  const nav2_util::LifecycleNode::WeakPtr & node,
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  const std::string & source_name,
+  const std::string & base_frame_id,
+  const tf2::Duration & transform_tolerance,
+  const tf2::Duration & data_timeout)
+: node_(node), tf_buffer_(tf_buffer), source_name_(source_name),
+  source_frame_id_(""), base_frame_id_(base_frame_id), fixed_frame_id_(""),
+  transform_tolerance_(transform_tolerance), data_timeout_(data_timeout)
+{
+  RCLCPP_INFO(logger_, "[%s]: Creating SourceBase", source_name_.c_str());
```

Comment:
Base class doesn't need the create / destroy logs; the implementations have them with more information

Reply:
Ditto.